### PR TITLE
Allergies exercise.

### PR DESCRIPTION
**Pull request by verdammelt**
Was a pretty straight-forward port of the existing Clojure exercise.

Would like to hear opinions on making it more (Common) Lispy.

### CODE REVIEW

**Comment by canweriotnow** [comment]
@verdammelt As you know, I'm more Clojurian than Common Lisper, but I think it looks effin' beautiful. I'm happy to merge if you are, or we could wait for @wobh to chime in :)

File: allergies/example.lisp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+(defun allergen (allergen-and-score) (first allergen-and-score))
+(defun score (allergen-and-score) (second allergen-and-score))
+
+(defun score-matches (allergen-and-score score)
+  (not (zerop (logand (score allergen-and-score) score))))
+
+(defun find-all (test sequence)
+  (loop for i in sequence
+     when (funcall test i) collecting i))
+
```
Comment (wobh) [inline allergies/example.lisp]: It's fine by me but, I'm a bit surprised you didn't `remove-if-not`.

**Comment by wobh** [comment]
If you want to change it, go ahead, otherwise, I'll merge later this evening (PDT).

**Comment by verdammelt** [comment]
`remove-if-not` would have been good. I think I got the `find-all` stuck in my head early and didn't think of anything else.

What nobody noticed is that the starter file has a bug :). I'll fix it shortly (on master).

**Comment by wobh** [comment]
Hah!

**Comment by verdammelt** [comment]
It was a sort of Freudian Slip I guess.

**Pull request merged**